diffkemp/simpll/DifferentialFunctionComparator.cpp

```diff
@@ -642,6 +642,57 @@ std::vector<SyntaxDifference> DifferentialFunctionComparator::findAsmDifference(
     return {diff};
 }
 
+/// Implement comparison of global values that does not use a
+/// GlobalNumberState object, since that approach does not fit the use case
+/// of comparing functions in two different modules.
+int DifferentialFunctionComparator::cmpGlobalValues(GlobalValue *L,
+                                                    GlobalValue *R) const {
+    int Result = L <= R;
+
+    if (L->hasName() && R->hasName() && Result) {
+        // Both values are named, compare them by names
+        auto NameL = L->getName();
+        auto NameR = R->getName();
+
+        // Remove number suffixes
+        if (hasSuffix(NameL))
+            NameL = NameL.substr(0, NameL.find_last_of("."));
+        if (hasSuffix(NameR))
+            NameR = NameR.substr(0, NameR.find_last_of("."));
+
+        if (NameL == NameR || (isPrintFunction(NameL) &&
+                isPrintFunction(NameR)))
+            Result = 0;
+    }
+
+    if (isa<GlobalVariable>(L) && isa<GlobalVariable>(R) && Result) {
+        // Global variables are compared using their initializers.
+        auto GVarL = dyn_cast<GlobalVariable>(L);
+        auto GVarR = dyn_cast<GlobalVariable>(R);
+        if (GVarL->hasInitializer() && GVarR->hasInitializer())
```

Comment:
That's a good idea, it actually wasn't done this way in DifferentialGlobalNumberState.

Reply:
I think it compared initializers in case the globals had the `unnamed_addr` flag set - this flags tells that the name is not important, only the value is. However, I think that we can generalize this to all constants since if two constants have the same value, they are semantically equal.